### DISCUSSION TITLE
Third party packages?

### DISCUSSION THREAD

**Discussion by robdekort** [Laravel-Lang/common#153] [Q&A] [answered]
Hi there!

First of all: what a great package this is. It really takes out the pain of adding and updating translation files. Thanks for that. 

I'm a heavy Statamic user and merging Laravel, Statamic and custom translations for the project you're working on still is a bit of a hassle. I'm not affiliated with Statamic but I was wondering if this package is open to merging in translations for third party packages like Statamic. If so I could open up an idea on their repo to see what they think about this. 

All the best,

Rob

**Comment by andrey-helldar** [comment, marked as answer]
Yes, it is possible and very easy.

In short, you need to collect all the keys for translation into one file, add the necessary locales to the package and then synchronize them.

Detailed tutorial using the [Translations Template](https://github.com/Laravel-Lang/translations-template).

For example, the new package will be named `best-packages/statamic-translations` and located in the GitHub project `BestPackages/StatamicTranslations`.

1. Copy the project to yourself in any way (`git clone`, `use this template` button, download archive...);
2. Replace `<your_namespace>` with `BestPackages/StatamicTranslations`;
3. Replace `your/namespace` with `best-packages/statamic-translations`;
4. Replace `YourNamespace\Translations` with `BestPackages\StatamicTranslations`;
5. Replace `Extended Lang Translations Template` with `Best Packages: Statamic Translations`;
6. Replace `<username>` with `Rob de Kort`;
7. Clean up `README.md` file;
8. Run the `composer update` console command;
9. Create the required localizations:
     - `vendor/bin/lang create nl nn` - for specific localizations
     - `vendor/bin/lang create` - will create all locales [supported](https://github.com/Laravel-Lang/locales/blob/1.9.0/src/Enums/Locale.php) by the Laravel Lang project.
10. Add Statamic project files either manually by copying them to the `source` folder, or create a GitHub Actions Rules like https://github.com/Laravel-Lang/lang/blob/main/.github/workflows/download. yml.
11. Run the console command `vendor/bin/lang sync`.
12. Remove unnecessary classes from the `src/Plugins` folder;
13. Create your own or use an existing plugin class in the `src/Plugins` folder, indicating in it the name of the package, if present, translations from the files described in the class should be installed. You can also specify the version for which this should be done. Or remove the override of the `$vendor` and `$version` parameters - in this case the translations will always be installed. An example can be seen [here](https://github.com/Laravel-Lang/lang/tree/main/src).
14. Don't forget to update or add a class reference in the [`src/Plugin.php`](https://github.com/Laravel-Lang/lang/blob/main/src/Plugin.php) file.
15. Publish the application on Packagist or use [repository links](https://getcomposer.org/doc/05-repositories.md#loading-a-package-from-a-vcs-repository) in the application.
16. Install the dependency in the application by calling the console command `composer require --dev best-packages/statamic-translations`.
17. Add the required localizations to the application. For example, `php artisan lang:add nn nl` or `php artisan lang:add` to install all locales.
16. In the application, call the `php artisan lang:update` command.
18. That's it.

**Reply by robdekort** [reply]
Thank you for your detailed answer Andrey. I'll let the Statamic core team know and see what they think. 🙏

**Reply by andrey-helldar** [reply]
If you have any questions, don't hesitate to contact us!)
